openmeter/subscription/service/update.go

```diff
@@ -30,343 +30,361 @@ func (s *service) Update(ctx context.Context, subscriptionID models.NamespacedID
 		return def, err
 	}
 
-	return s.sync(ctx, view, newSpec)
+	return transaction.Run(ctx, s.TransactionManager, func(ctx context.Context) (subscription.Subscription, error) {
+		sub, err := s.sync(ctx, view, newSpec)
+		if err != nil {
+			return sub, err
+		}
+
+		// Let's fetch the view for event generation
+		view, err := s.GetView(ctx, sub.NamespacedID)
+		if err != nil {
+			return sub, err
+		}
+
+		err = s.Publisher.Publish(ctx, subscription.UpdatedEvent{
+			UpdatedView: view,
+		})
+		if err != nil {
+			return sub, fmt.Errorf("failed to publish event: %w", err)
+		}
+
+		return sub, nil
+	})
 }
 
 // TODO: localize error so phase and item keys are always included (alongside subscription reference)
 func (s *service) sync(ctx context.Context, view subscription.SubscriptionView, newSpec subscription.SubscriptionSpec) (subscription.Subscription, error) {
-	return transaction.Run(ctx, s.TransactionManager, func(ctx context.Context) (subscription.Subscription, error) {
-		var def subscription.Subscription
+	var def subscription.Subscription
```

Comment:
please keep the contents of sync inside a transaction block, it shouldn't expect the caller to guarantee it

Reply:
also, this shows a diff for almost the entire file, was there any actual difference in this logic?

Reply:
> please keep the contents of sync inside a transaction block, it shouldn't expect the caller to guarantee it

Ok, I have changed it, but I don't see the logic here, as other call sites properly wrap it into a transaction. I hoped that the consistency here is that external calls are doing the wrapping.

> also, this shows a diff for almost the entire file, was there any actual difference in this logic?

Please enable the ignore the whitespace changes when reviewing. That helps with ignoring the unneeded changes. In this specific setup I have moved the trns block to the call sites as the other 2 places it was invoked from was already in trns.

Please re-check the PR.